src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -103,70 +122,113 @@
     String getId();
 
     /**
-     * Gets a short description of the linked permission. May include a link to
-     * a more detailed description on the plugin's web page.
+     * Gets a short description of the linked permission.
+     *
+     * <p>May include a link to a more detailed description on the plugin's
+     * web page.</p>
+     *
+     * <p>Will return an empty optional for descriptions which have been
+     * automatically generated, or where a description was omitted when the
+     * {@link PermissionDescription} was created.</p>
      *
      * @return A short description of the linked permission
      */
-    Text getDescription();
+    Optional<Text> getDescription();
 
     /**
-     * Gets all subjects that have this permission set.
+     * Gets the owning plugin the permission belongs to.
+     *
+     * <p>Will return an empty optional for descriptions which have been
+     * automatically generated.</p>
+     *
+     * @return The owning plugin the permission belongs to
+     */
+    Optional<PluginContainer> getOwner();
```

Comment:
Just noticed this.

Can you elaborate on how this can be Optional? No matter what is executing code, a PluginContainer is involved (even for the base game).